lib/rack/sendfile.rb

```diff
@@ -117,7 +118,7 @@ def call(env)
           path = ::File.expand_path(body.to_path)
           if url = map_accel_path(env, path)
             headers[CONTENT_LENGTH] = '0'
-            headers[type] = url
+            headers[type] = WEBrick::HTTPUtils.escape(url)
```

Comment:
Does `Rack::Utils.escape` work here?

Reply:
No, since ``Rack::Utils.escape`` escapes the URL as a whole (i.e. it also escapes the frontslashes, colons etc.) whereas ``WEBrick::HTTPUtils.escape`` keeps the URL intact and only escapes the characters necessary:

```ruby
2.5.3 :001 > require 'rack'
2.5.3 :002 > require 'webrick'
2.5.3 :003 > url = 'http://test.com/hello%/world'
2.5.3 :004 > Rack::Utils.escape(url)
 => "http%3A%2F%2Ftest.com%2Fhello%25%2Fworld" 
2.5.3 :005 > WEBrick::HTTPUtils.escape(url)
 => "http://test.com/hello%25/world" 
```

Reply:
Thank you for your comments! 😃

It seems that nginx decodes whatever percent-encoded characters, so encoding '/' to '%2F' is not a problem. That works fine.
Instead, there is another problem that `Rack::Utils.escape` encodes ' ' (space) to '+'. Because nginx does not decode '+' to ' ', `Rack::Utils.escape` does not work if a path includes space characters.
`WEBrick::HTTPUtils.escape` works fine in the both cases.

Reply:
We should not be using `WEBrick` though in rack. It is not a dependency of the library and I don't think it should be since users may opt out to a different web server.

Reply:
Isn't `WEBrick` part of the standard library/default gems in ruby?

Reply:
Yes, but is not required by default. You can only use it if you want. If use in rack you will have to load it, and I want to avoid this.

Reply:
How about using ``URI#escape`` then? I quickly tested it in irb and it seems to provide the same functionality:

```ruby
2.5.3 :001 > require 'uri'
 => true 
2.5.3 :002 > URI.escape('http://test.com/hello%/world')
 => "http://test.com/hello%25/world"
````

I'm not sure about availability in different ruby versions, though.